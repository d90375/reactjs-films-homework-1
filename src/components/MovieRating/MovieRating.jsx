import React from 'react';
import PropTypes from 'prop-types';

import styles from './MovieRating.scss';

const MovieRating = ({ rating }) => {
  const stars = Array.from(Array(5), (star, index) => {
    if (Math.round(rating / 2 - (index + 1)) >= 0) {
      return <span className={styles.star} key={index}>&#9733;</span>;
    }

    return <span className={styles.star} key={index}>&#9734;</span>;
  });

  return (
    <div className={styles.container}>
      { stars }
      <span className={styles.rating}>{rating}</span>
    </div>
  );
};

MovieRating.propTypes = {
  rating: PropTypes.number.isRequired,
};

export default MovieRating;
