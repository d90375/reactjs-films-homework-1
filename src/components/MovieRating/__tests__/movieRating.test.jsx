import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import MovieRating from '../MovieRating';

const rating = '3';

test('MovieRating renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<MovieRating rating={rating} />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
