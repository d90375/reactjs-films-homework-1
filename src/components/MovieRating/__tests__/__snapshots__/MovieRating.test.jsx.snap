// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieRating renders correctly 1`] = `
<div
  className="container"
>
  <span
    className="star"
  >
    ★
  </span>
  <span
    className="star"
  >
    ★
  </span>
  <span
    className="star"
  >
    ★
  </span>
  <span
    className="star"
  >
    ☆
  </span>
  <span
    className="star"
  >
    ☆
  </span>
  <span
    className="rating"
  >
    3
  </span>
</div>
`;
